app/frontend/wallet/helpers/CachedDeriveXpubFactory.ts

```diff
@@ -28,13 +37,43 @@ function CachedDeriveXpubFactory(derivationScheme, deriveXpubHardenedFn) {
     return await derivedXpubs[memoKey]
   }
 
-  async function deriveXpubNonhardenedFn(derivationPath) {
+  async function deriveXpubNonhardenedFn(derivationPath: BIP32Path) {
     const lastIndex = derivationPath.slice(-1)[0]
     const parentXpub = await deriveXpub(derivationPath.slice(0, -1))
-
     return deriveChildXpub(parentXpub, lastIndex, derivationScheme.ed25519Mode)
   }
 
+  function createPathBulk(derivationPath: BIP32Path): BIP32Path[] {
+    const paths: BIP32Path[] = []
+    const accountIndex = derivationPath[2] - HARDENED_THRESHOLD
+    const baseAccountIndex = accountIndex - (accountIndex % MAX_BULK_EXPORT_AMOUNT)
```

Comment:
I'd rather introduce the abstraction of "currentAccountPage" which would be the index of the current "batch" of addresses and it would be accountIndex / MAX_BULK_EXPORT_AMOUNT

Reply:
Thanks, great idea. :+1: